enterprise/frontend/src/metabase-enterprise/clean_up/CleanupCollectionModal/CleanupCollectionBulkActions.tsx

```diff
@@ -0,0 +1,44 @@
+import { msgid, ngettext, t } from "ttag";
+import _ from "underscore";
+
+import {
+  BulkActionBar,
+  BulkActionButton,
+} from "metabase/components/BulkActionBar";
+import type { CollectionItem } from "metabase-types/api";
+
+import CS from "./CleanupCollectionBulkActions.module.css";
+
+interface CleanupCollectionBulkActionsProps {
+  selected: CollectionItem[];
+  clearSelectedItem: () => void;
+}
+
+export const CleanupCollectionBulkActions = ({
+  selected,
+  clearSelectedItem,
+}: CleanupCollectionBulkActionsProps) => {
+  const actionMessage = ngettext(
+    msgid`${selected.length} item selected`,
+    `${selected.length} items selected`,
+    selected.length,
+  );
+
+  const handleBulkArchive = async () => {
+    const actions = selected.map(item => item?.setArchived?.(true));
+    Promise.all(actions).finally(() => clearSelectedItem());
```

Comment:
I didn't know about `finally` - neat! A bit more idiomatic, maybe worth considering: `await Promise.all(actions); clearSelectedItem()`

Reply:
But what you have is great

Reply:
`.finally` will run wether the promise succeeds or fails, but awaiting a failed promise w/o catching would break the control flow and never execute `clearSelectedItem`